src/frontend/get-jira-configuration.ts

```diff
@@ -57,27 +62,30 @@ interface FailedConnections {
 	orgName: string | undefined;
 }
 
-export const getFailedConnections = (
+export const getFailedConnections = async (
 	installations: FailedInstallations[],
 	subscriptions: SubscriptionClass[]
-): FailedConnections[] => {
-	return installations
+): Promise<FailedConnections[]> => {
+	return Promise.all(installations
 		.filter((response) => !!response.error)
-		.map((failedConnection: FailedInstallations) => {
-			const sub = subscriptions.find(
-				(subscription: SubscriptionClass) =>
-					failedConnection.id === subscription.gitHubInstallationId
-			);
-			const repos = sub?.repoSyncState?.repos || {};
-			const repoId = Object.keys(repos);
-			const orgName = repos[repoId[0]]?.repository?.owner.login || undefined;
+		.map(async (failedConnection: FailedInstallations) => {
+			const sub = subscriptions.find(sub => failedConnection.id === sub.gitHubInstallationId);
+			let orgName;
+			if (sub && await booleanFlag(BooleanFlags.REPO_SYNC_STATE_AS_SOURCE, false, sub.jiraHost)) {
+				const repo = await RepoSyncState.findOneFromSubscription(sub);
+				orgName = repo.repoOwner;
+			} else {
+				const repos = sub?.repoSyncState?.repos || {};
```

Comment:
At this point would sub already have repoSyncState retrieved from DB? 

If yes, how does our flag prevents us from retrieving it from the old table?

Reply:
it doesn't.  I'm not sure I understand why it shouldn't.  We should be able to flip between the feature flag on/off without any data loss or issues.  The next PR, after this one is 100%, will remove the `repoSyncState` column altogether which will now reduce the DB load considerably.

Reply:
Cool, so this PR is to make a flag which changes the logic to not use `repoSyncState` anymore and then in another PR we'll stop fetching it when the flag is rolled out.

Am I understand it correctly?

Reply:
Correct :)